2025-09-06 10:12:00,723 ERROR: Exception on / [GET] [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask/app.py:875]
Traceback (most recent call last):
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 951, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: no such table: user

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask/app.py", line 915, in full_dispatch_request
    rv = self.preprocess_request()
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask/app.py", line 1291, in preprocess_request
    rv = self.ensure_sync(before_func)()
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/routes.py", line 13, in before_request
    if current_user.is_authenticated:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/werkzeug/local.py", line 318, in __get__
    obj = instance._get_current_object()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/werkzeug/local.py", line 526, in _get_current_object
    return get_name(local())
                    ^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask_login/utils.py", line 25, in <lambda>
    current_user = LocalProxy(lambda: _get_user())
                                      ^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask_login/utils.py", line 370, in _get_user
    current_app.login_manager._load_user()
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/flask_login/login_manager.py", line 364, in _load_user
    user = self._user_callback(user_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/models.py", line 103, in load_user
    return db.session.get(User, int(id))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/scoping.py", line 1058, in get
    return self._proxied.get(
           ^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 3694, in get
    return self._get_impl(
           ^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 3873, in _get_impl
    return db_load_fn(
           ^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/loading.py", line 695, in load_on_pk_identity
    session.execute(
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
             ^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1419, in execute
    return meth(
           ^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 526, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1641, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 2355, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/.venv/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 951, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: user
[SQL: SELECT user.id AS user_id, user.username AS user_username, user.email AS user_email, user.password_hash AS user_password_hash, user.about_me AS user_about_me, user.last_seen AS user_last_seen 
FROM user 
WHERE user.id = ?]
[parameters: (1,)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
2025-09-06 10:13:32,809 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:13:41,008 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:13:45,310 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:28:33,582 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:29:23,233 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:29:32,614 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:29:51,642 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:30:41,275 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:30:41,278 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:30:41,278 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:30:44,477 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:31:20,197 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:31:46,884 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:32:14,321 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:36:11,157 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:38:49,035 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:39:40,874 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:39:40,877 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:39:40,877 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
2025-09-06 10:39:43,530 INFO: Microblog startup [in /home/ursmueller/programmieren/python/flask/tutorials/microblog/microblog/app/__init__.py:52]
